be/src/storage/chunk_helper.cpp

```diff
@@ -833,14 +833,14 @@ std::vector<ColumnPtr> SegmentedColumn::columns() const {
         return _cached_columns;
     }
     std::vector<ColumnPtr> columns;
-    for (auto& segment : _chunk->segments()) {
+    for (auto& segment : _chunk.lock()->segments()) {
```

Comment:
Do we need to handle lock failures?

Reply:
no. the `SegmentedColumn` is always held by `SegmentedChunk`, so it would always exist